api/v1alpha1/alluxiocluster_types.go

```diff
@@ -257,14 +257,24 @@ type AlluxioMonitorSpec struct {
 type EtcdSpec struct {
 	Enabled      *bool             `json:"enabled" yaml:"enabled"`
 	Auth         EtcdAuthSpec      `json:"auth,omitempty" yaml:"auth,omitempty"`
-	Image        EtcdImageSpec     `json:"image,omitempty" yaml:"image,omitempty"`
 	NodeSelector map[string]string `json:"nodeSelector,omitempty" yaml:"nodeSelector,omitempty"`
 	ReplicaCount int               `json:"replicaCount,omitempty" yaml:"replicaCount,omitempty"`
+	Resources    ResourcesSpec     `json:"resources,omitempty" yaml:"resources,omitempty"`
 }
 
 type EtcdAuthSpec struct {
-	Rbac  EtcdAuthRbacSpec  `json:"rbac,omitempty" yaml:"rbac,omitempty"`
-	Token EtcdAuthTokenSpec `json:"token,omitempty" yaml:"token,omitempty"`
+	Client EtcdAuthClientSpec `json:"client,omitempty" yaml:"client,omitempty"`
+	Peer   EtcdAuthPeerSpec   `json:"peer,omitempty" yaml:"peer,omitempty"`
+	Rbac   EtcdAuthRbacSpec   `json:"rbac,omitempty" yaml:"rbac,omitempty"`
+	Token  EtcdAuthTokenSpec  `json:"token,omitempty" yaml:"token,omitempty"`
+}
+
+type EtcdAuthClientSpec struct {
+	EnableAuthentication *bool `json:"enableAuthentication,omitempty" yaml:"enableAuthentication,omitempty"`
```

Comment:
are we using the bool pointer bcos we want to do deepcopy? are these fields by default immutable after intializing?

Reply:
No. It's because the bool value false is omitted in json marshal, which is not good...